src/components/Carousel.tsx

```diff
@@ -1,18 +1,171 @@
 import React from 'react';
 import './Carousel.scss';
+import { Input } from './FormFields';
 
-const Carousel: React.FC = () => (
-  <div className="Carousel">
-    <ul className="Carousel__list">
-      <li><img src="./img/1.png" alt="1" /></li>
-      <li><img src="./img/1.png" alt="2" /></li>
-      <li><img src="./img/1.png" alt="3" /></li>
-      <li><img src="./img/1.png" alt="4" /></li>
-    </ul>
-
-    <button type="button">Prev</button>
-    <button type="button">Next</button>
-  </div>
-);
+type Props = {
+  images: string[],
+};
+
+type State = {
+  itemWidth: number,
+  frameSize: number,
+  step: number,
+  animationDuration: number,
+  translate: number,
+};
+
+class Carousel extends React.Component<Props, State> {
+  state = {
+    itemWidth: 130,
+    frameSize: 3,
+    step: 3,
+    animationDuration: 1000,
+    translate: 0,
+  };
+
+  stepPrev = () => {
+    const { step, itemWidth, translate } = this.state;
+
+    if (translate + step * itemWidth < 0) {
+      this.setState(prevState => ({
+        translate: prevState.translate + step * itemWidth,
+      }));
+    } else {
+      this.setState({ translate: 0 });
+    }
+  };
+
+  stepNext = () => {
+    const {
+      step, itemWidth, translate, frameSize,
+    } = this.state;
+
+    const hiddenScroll = -(
+      itemWidth * this.props.images.length - frameSize * itemWidth);
+
+    if (translate - step * itemWidth < hiddenScroll) {
+      this.setState({ translate: hiddenScroll });
+    } else {
+      this.setState((prevState) => ({
+        translate: prevState.translate - step * itemWidth,
+      }));
+    }
+  };
+
+  widthChanger = (event: React.ChangeEvent<HTMLInputElement>) => {
+    this.setState({ itemWidth: +event.target.value });
+  };
+
+  stepChanger = (event: React.ChangeEvent<HTMLInputElement>) => {
+    this.setState({ step: +event.target.value });
+  };
+
+  AnimationDurationChanger = (event: React.ChangeEvent<HTMLInputElement>) => {
```

Comment:
camelCase

```suggestion
  animationDurationChanger = (event: React.ChangeEvent<HTMLInputElement>) => {
```
CamelCase might be used only for Components/Classes/Types